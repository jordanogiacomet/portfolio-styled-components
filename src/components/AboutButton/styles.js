import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
