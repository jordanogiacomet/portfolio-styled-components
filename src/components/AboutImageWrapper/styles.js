import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
