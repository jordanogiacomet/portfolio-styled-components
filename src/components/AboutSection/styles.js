import styled from 'styled-components';

export const Container = styled.section`
  padding: 50px 0;
  display: flex;
  flex-direction: column;

  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }

  @media (min-width: 1280px) {
    gap: 50px;
  }
`;

