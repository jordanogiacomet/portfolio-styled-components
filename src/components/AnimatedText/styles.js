import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: black;
  }
  &.dark{
    color: white;
  }
`;