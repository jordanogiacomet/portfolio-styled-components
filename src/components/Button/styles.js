import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 