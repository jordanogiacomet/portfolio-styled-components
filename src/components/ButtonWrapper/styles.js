import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;