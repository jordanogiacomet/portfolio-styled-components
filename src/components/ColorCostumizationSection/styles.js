// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
