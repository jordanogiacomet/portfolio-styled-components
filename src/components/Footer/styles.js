import styled from 'styled-components';

export const Container = styled.footer`
  border-top: 1px solid #33353F;
  border-left: transparent;
  border-right: transparent;
  z-index: 10;
  color: white;
  width: 100%;
  background-color: #18191E;

  &.light {
    background-color: #1E3A8A;
  }
`;

export const Wrapper = styled.div`
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (max-width: 640px) {
    flex-direction: column;
    gap: 20px;
  }
`;

export const Text = styled.p`
  text-align: center;
  margin-top: 2rem;
  color: #ADB7BE;

  &.light {
    color: white;
  }
`;

export const Logo = styled.span`
  font-size: 1.5rem;
  font-weight: bold;
`;
