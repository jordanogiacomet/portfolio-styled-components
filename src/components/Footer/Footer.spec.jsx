import { screen } from '@testing-library/react';
import { Footer } from '.';
import '@testing-library/jest-dom/extend-expect';
import { renderTheme } from '../../styles/render-theme';
import { ColorProvider } from '../../contexts/ColorCustomizationContext';
import { ColorModeProvider } from '../../contexts/ColorModeContext';
import { LanguageProvider } from '../../contexts/LanguageContext';

describe('<Footer />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <Footer />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render logo link', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <Footer />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    const logoLink = screen.getByRole('link', { name: 'LOGO' });
    expect(logoLink).toBeInTheDocument();
  });
});
