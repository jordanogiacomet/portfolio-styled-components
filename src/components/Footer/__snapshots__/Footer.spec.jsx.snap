// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
.c0 {
  border-top: 1px solid #33353F;
  border-left: transparent;
  border-right: transparent;
  z-index: 10;
  color: white;
  width: 100%;
  background-color: #18191E;
}

.c0.light {
  background-color: #1E3A8A;
}

.c1 {
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c3 {
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;
}

.c3 >img {
  max-height: 3rem;
}

.c2 {
  color: #ADB7BE;
  font-size: 2.4rem;
  text-transform: uppercase;
  z-index: 2;
}

.c4 {
  padding: 1rem;
  background-color: transparent;
  box-shadow: none;
  border: 2px solid #ADB7BE;
  border-radius: 9999px;
  color: #ADB7BE;
  font-size: 1.2rem;
  right: 2.5rem;
  margin-right: 2rem;
  cursor: pointer;
  transition: 0.2s;
}

.c4.light {
  color: white;
  border-color: white;
}

.c4.light:hover {
  color: #CCCCCC;
  border-color: #CCCCCC;
}

.c4:hover {
  color: #4169E1;
  border-color: #4169E1;
}

.c5 {
  padding: 1rem;
  background-color: transparent;
  box-shadow: none;
  border: 2px solid #ADB7BE;
  border-radius: 9999px;
  color: #ADB7BE;
  font-size: 1.2rem;
  right: 10rem;
  cursor: pointer;
  z-index: 199999;
  transition: 0.2s;
}

.c5:hover {
  color: #4169E1;
  border-color: #4169E1;
}

.c5.light {
  color: white;
  border-color: white;
}

.c5.light:hover {
  color: #CCCCCC;
  border-color: #CCCCCC;
}

@media (max-width: 640px) {
  .c1 {
    flex-direction: column;
    gap: 20px;
  }
}

@media (max-width: 768px) {
  .c4 {
    bottom: 20px;
  }
}

@media (max-width: 768px) {
  .c5 {
    bottom: 20px;
  }
}

<div>
  <footer
    class="c0 dark"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          <span>
            LOGO
          </span>
        </a>
      </h1>
      <div>
        <button
          aria-label="Switch language"
          class="c4 dark"
        >
          PT-BR
        </button>
        <button
          aria-label="Switch to light mode"
          class="c5 dark"
        >
          DARK
        </button>
      </div>
    </div>
  </footer>
</div>
`;
